Globals/MAPCUSTODIAN.gd

```diff
@@ -0,0 +1,92 @@
+extends Node
+
+# This script handles loading maps from files and storing them for game access
+var mapPathArray = []
+var _curMapIdx = 0
+var masterMapArray = []
+export var mapsPath = 'res://Maps/'
+
+
+func _ready():
+	reloadMaps()
+	var mapCode = validateMaps()
+	if mapCode == '0':
+		pass
+	else:
+		print_debug(mapCode)
+		get_tree().quit()
+
+
+func reloadMaps():
+	mapPathArray = []
+	var mapDirectory = Directory.new()
+	mapDirectory.open(mapsPath)
+	mapDirectory.list_dir_begin()
+	while true:
+		var folder = mapDirectory.get_next()
+		if folder == "":
+			break
+		elif not folder.begins_with("."):
+			mapPathArray.append(folder)
+
+func validateMaps():
+	var fileChecker = File.new()
+	for d in mapPathArray:
+		var mapDataPath = mapsPath + d + '/mapData.json'
+		var terrainDataPath = mapsPath + d + '/terrainData.csv'
+		var unitDataPath = mapsPath + d + '/unitData.json'
+		# Check for each of the required files per map - ensure they exist
+		print_debug("Checking map: " + d)
+		if not fileChecker.file_exists(mapDataPath):
+			return ("Unable to find " + mapDataPath)
+		if not fileChecker.file_exists(terrainDataPath):
+			return ("Unable to find " + terrainDataPath)
+		if not fileChecker.file_exists(unitDataPath):
+			return ("Unable to find " + unitDataPath)
+		#fileChecker.open(mapDataPath)
+		var errors = validateMapData(mapDataPath)
+		if errors != '0':
+			return errors
+		errors = validateTerrainData(terrainDataPath)
+		if errors != '0':
+			return errors
+		errors = validateUnitData(unitDataPath)
+		
+		# At this point we have validated all of the data within the files
+		# So we can officially load the map into the master map array for use
+		if errors == '0':
+			var goodMap = load('res://Testing/Map.gd').new(mapsPath + d)
+			masterMapArray.append(goodMap)
+		return errors
+
+
+# TODO: Actually validate the data - we are just pretending right now and dealing with the crash
+func validateMapData(filepath):
```

Comment:
We can do it in a future PR but we should definitely make these enums or something to key off of instead of just chars.

Reply:
Agreed